fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -1286,17 +1307,38 @@ struct TenantManagementWorkload : TestWorkload {
 			try {
 				if (self->useMetacluster) {
 					state std::vector<std::pair<TenantName, metacluster::MetaclusterTenantMapEntry>> metaTenants =
-					    wait(metacluster::listTenantMetadata(self->mvDb, beginTenant, endTenant, limit));
+					    wait(metacluster::listTenantMetadata(self->mvDb,
+					                                         beginTenant,
+					                                         endTenant,
+					                                         limit,
+					                                         /*offset=*/0,
+					                                         /*filters=*/std::vector<metacluster::TenantState>(),
+					                                         tGroup));
 					verifyTenantList<metacluster::MetaclusterTenantMapEntry>(
-					    self, metaTenants, limit, beginTenant, endTenant);
+					    self, metaTenants, limit, beginTenant, endTenant, tGroup);
 				} else {
 					state std::vector<std::pair<TenantName, TenantMapEntry>> tenants =
 					    wait(listTenantsImpl(tr, beginTenant, endTenant, limit, operationType, self));
 					if (operationType == OperationType::METACLUSTER) {
 						ASSERT_EQ(tenants.size(), 0);
 						return Void();
 					}
-					verifyTenantList<TenantMapEntry>(self, tenants, limit, beginTenant, endTenant);
+					verifyTenantList<TenantMapEntry>(
+					    self, tenants, limit, beginTenant, endTenant, Optional<TenantGroupName>());
+
+					// Get all tenants in the specified tenant group
+					// In a non-metacluster, the return type of the filtered tenants is slightly different
+					// which makes it messy to bundle above. Instead, verify it separately.
```

Comment:
The issue with doing this separately is that this is a weaker version of the check above. Since it doesn't look like we are using any properties of the `TenantMapEntry*` in the list verification, maybe we could just keep a set of the tenant names and pass those into the verification function, and that would work regardless of what type we get. Or we could pass in the map with the second type of the pair templated so that `int64_t` or `TenantMapEntry` doesn't matter.

Reply:
I ended up trying to split it up into two different cases. The problem with all the choices in `listTenantImpl` is that those calls can't specify the tenant groups. Instead I put it into its own case if a tenant group was present and ignored the operation type picked by the test.